sdmetrics/reports/single_table/_properties/synthesis.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+import pandas as pd
+import plotly.express as px
+
+from sdmetrics.reports.single_table._properties import BaseSingleTableProperty
+from sdmetrics.single_table import NewRowSynthesis
+
+
+class Synthesis(BaseSingleTableProperty):
+    """Synthesis property class for single table.
+
+    This property assesses the novelty of the syntetic data over the real data.
+    The ``NewRowSynthesis`` metric is computed over the real and synthetic table to
+    score the proportion of new rows in the synthetic data.
+    """
+
+    metric = NewRowSynthesis
+
+    def _generate_details(self, real_data, synthetic_data, metadata, progress_bar=None):
+        """Generate the _details dataframe for the synthesis property.
+
+        Args:
+            real_data (pandas.DataFrame):
+                The real data
+            synthetic_data (pandas.DataFrame):
+                The synthetic data
+            metadata (dict):
+                The metadata of the table
+            progress_bar (tqdm.tqdm or None):
+                The progress bar to use. Defaults to tqdm.
+
+        Returns:
+            pandas.DataFrame
+        """
+        name = self.metric.__name__
+        error_message = np.nan
+        if len(synthetic_data) > 10000:
+            sample_size = 10000
+        else:
+            sample_size = len(synthetic_data)
+
+        try:
+            score_breakdown = self.metric.compute_breakdown(
+                real_data, synthetic_data, synthetic_sample_size=sample_size
+            )
+            score = score_breakdown['score']
+            num_matched_rows = score_breakdown['num_matched_rows']
+            num_new_rows = score_breakdown['num_new_rows']
+
+        except Exception as e:
+            score = np.nan
+            num_matched_rows = np.nan
+            num_new_rows = np.nan
+            error_message = f'Error: {type(e).__name__} {e}'
+
+        finally:
+            if progress_bar:
+                progress_bar.update()
+
+        result = pd.DataFrame({
```

Comment:
Are any of the objects a list? If not, it's weird to me we are using a DataFrame when a simple dict would suffice.

Reply:
It's to keep the same abstraction as the other properties. `_generate_details` should generate the details table, which is a DataFrame. I agree it's a bit weird because there is only 1 row.